lib/core/shared/sdk/funnelProtocol.js

```diff
@@ -32,16 +32,18 @@ class FunnelProtocol extends KuzzleEventEmitter {
     super();
 
     this.id = 'funnel';
-    this.user = user;
-
     this.requestOptions = null;
+    this._impersonateKuid = null;
 
-    if (this.user && typeof this.user._id !== 'string') {
-      throw kerror.get('plugin', 'context', 'invalid_user');
+    if (user) {
```

Comment:
The `FunnelProtocol` should not be instantiated with a user object anymore. It's a leftover from the previous implementation of the `as` method where the `EmbeddedSDK` could be instantiated with an user.

Now the `EmbeddedSDK` should never be instantiated with an user since we have the `ImpersonatedSDK` for this.

You should also remove the `user` argument from the `EmbeddedSDK` constructor

Reply:
Oh ofc, getting that user in the funnel's constructor actually induced some underlying logic complexities. I thought we couldn't remove it due to 'trying to maintain backwards compatibility'.